Drive/rover_drive_system.hpp

```diff
@@ -51,37 +52,48 @@ class RoverDriveSystem
     SetMode();
   }
 
-  // Retrieves /drive Mission Control commands.
+  // Handles GET /drive?parameters for rover drive system to mission control
   /// @return returns true if connection is established from mission control
   bool GetMissionControlData()
   {
-    // GET /drive JSON value ?
+    mission_control_data_.http_GET_param =
+        "?is_operational=%d&drive_mode=%c&left_wheel_speed=%f&right_wheel_"
+        "speed=%f&back_wheel_speed=%f",
+    mission_control_data_.is_operational, static_cast<char>(current_mode_),
+    left_wheel_.GetSpeed(), right_wheel_.GetSpeed(), back_wheel_.GetSpeed();
+    // TODO: GET /drive?key=value&key=value... JSON value. Refer to
+    // GetRoverData() to find GET paramater
     if (true)  // change 'true' to something for localhost connection found
     {
       char response[] = {};
       ParseMissionControlData(response);
-      Move(mission_control_data_.drive_mode,
-           mission_control_data_.rotation_angle, mission_control_data_.speed);
       return true;
     }
-    sjsu::LogError("Unable to reach mission control server");
-    return false;
+    else
+    {
+      SetMode();
+      sjsu::LogError("Unable to reach mission control server");
+      return false;
+    }
   }
 
   /// Main function for handling all the rover drive system functionality.
-  /// @param mode Updates drive mode if new mode is different from current.
-  /// @param rotation_angle adjusts wheel position depending on mode.
-  /// @param speed adjusts the movement speed of the rover.
-  void Move(char mode, float rotation_angle, float speed)
+  /// drive_mode updates drive mode if new mode is different.
+  /// rotation_angle adjusts wheel position depending on mode.
+  /// speed adjusts the movement speed of the rover.
+  void Move()
   {
     if (mission_control_data_.is_operational)
     {
-      if (mode != static_cast<char>(current_mode_))
+      if (mission_control_data_.drive_mode != static_cast<char>(current_mode_))
       {
-        SetMode(mode);
+        SetMode(mission_control_data_.drive_mode);
```

Comment:
Shouldn't you store the new `drive_mode` into `current_mode_` so you don't end up looping this forever?

Reply:
SetMode() will update current_mode_ with the new drive mode